src/components/editor/metaform-editor-right-drawer-feature.tsx

```diff
@@ -0,0 +1,455 @@
+import { Button, Checkbox, Divider, FormControl, FormControlLabel, InputLabel, MenuItem, Select, Stack, TextField, Typography } from "@mui/material";
+import { Metaform, MetaformField, MetaformFieldOption, MetaformSection, MetaformTableColumn, MetaformTableColumnType } from "generated/client";
+import produce from "immer";
+import slugify from "slugify";
+import strings from "localization/strings";
+import React, { useEffect, FC } from "react";
+import DeleteIcon from "@mui/icons-material/Delete";
+
+/**
+ * Component properties
+ */
+interface Props {
+  sectionIndex?: number;
+  fieldIndex?: number;
+  pendingForm: Metaform;
+  setPendingForm: (metaform: Metaform) => void;
+}
+
+/**
+ * Draft editor right drawer feature component
+ */
+const MetaformEditorRightDrawerFeatureComponent: FC<Props> = ({
+  sectionIndex,
+  fieldIndex,
+  pendingForm,
+  setPendingForm
+}) => {
+  const [ metaformSectionTitle, setMetaFormSectionTitle ] = React.useState<string | undefined>("");
+  const [ columnType, setColumnType ] = React.useState<string>("");
+
+  /**
+   * Set values of Confidition field, switch and get selected component name
+   */
+  const setVisiblityComponentValues = () => {
+    if (fieldIndex === undefined && sectionIndex !== undefined) {
+      setMetaFormSectionTitle(pendingForm.sections![sectionIndex].title ?? "");
+    }
+  };
+
+  /**
+   * Updates metaform field
+   * @param metaformField Metaform field what we are editing
+   */
+  const updateFormField = (metaformField: MetaformField) => {
+    if (sectionIndex === undefined || fieldIndex === undefined) {
+      return;
+    }
+    const updatedForm = produce(pendingForm, draftForm => {
+      draftForm.sections?.[sectionIndex]?.fields?.splice(fieldIndex, 1, metaformField);
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Updates metaform section
+   * @param metaformSection metaform section what we are editing
+   */
+  const updateFormSection = (metaformSection: MetaformSection) => {
+    if (sectionIndex === undefined) {
+      return;
+    }
+    const updatedForm = produce(pendingForm, draftForm => {
+      draftForm.sections?.splice(sectionIndex, 1, metaformSection);
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+  * Renders fields category title
+  * @param title Category title
+  */
+  const renderFieldsCategoryTitle = (title: string) => {
```

Comment:
Is this method necessary? Naming is bit confusing, what category?

Reply:
Deleted whole method